lib/terrafying/components/letsencrypt.rb

```diff
@@ -324,15 +291,54 @@ def renew(name, bucket, domains, options={})
                         "route53:ChangeResourceRecordSets",
                       ],
                       Resource:
-                        domains.map { | domain |
-                          "arn:aws:route53:::#{domain.zone.id[1..-1]}"
+                        @zones.reject(&:nil?).map { | zone |
+                          "arn:aws:route53:::#{zone.id[1..-1]}"
                         },
                       Effect: "Allow"
                     }
                   ]
                 }
               )
             }
+          end
+        super
+      end
+
+      def renew
+        resource :aws_lambda_function, "#{@name}_lambda", {
+          function_name: "#{@name}_lambda",
+          s3_bucket: "uswitch-certbot-lambda",
+          s3_key: "certbot-lambda.zip",
+          handler: "main.handler",
+          runtime: "python3.7",
+          timeout: "900",
+          role: "${aws_iam_role.#{@name}_lambda_execution.arn}",
```

Comment:
If you move the iam_role definition to above this, you can do the following:

```
role = resource :aws_iam_role, ... { 
  ...
}

resource :aws_lambda_function, ..., {
  roles: role["arn"],
}
```

The output of the resource function gives you something you can interact with sort of like a hash and it will generate the interoplation values. Much more rubyish than embedding terraform strings.